### PR TITLE
pdms: PD default using start script v2 with WaitForDnsNameIpMatch when enables PDMS

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Ref https://github.com/tikv/pd/issues/7519 , https://github.com/pingcap/tidb-operator/pull/5381

PD&PDMS use start script v2 and enabling feature `WaitForDnsNameIpMatch` when enabled PDMS

need to check IP firstly
```
$ kubectl logs -f basic-pd-0 -n pingcap
domain resolve basic-pd-0.basic-pd-peer.pingcap.svc success
10.244.0.64
hostIps: 10.244.0.69
resolvedIps: 10.244.0.64
Resolved IP does not match any of podIPs
domain resolve basic-pd-0.basic-pd-peer.pingcap.svc success
10.244.0.64
hostIps: 10.244.0.69
resolvedIps: 10.244.0.64
Resolved IP does not match any of podIPs
domain resolve basic-pd-0.basic-pd-peer.pingcap.svc success
10.244.0.64
hostIps: 10.244.0.69
resolvedIps: 10.244.0.64
Resolved IP does not match any of podIPs
domain resolve basic-pd-0.basic-pd-peer.pingcap.svc success
...
10.244.0.64
hostIps: 10.244.0.69
resolvedIps: 10.244.0.64
Resolved IP does not match any of podIPs
domain resolve basic-pd-0.basic-pd-peer.pingcap.svc success
domain resolve basic-pd-0.basic-pd-peer.pingcap.svc success
10.244.0.69
hostIps: 10.244.0.69
resolvedIps: 10.244.0.69
Success: Resolved IP matches one of podIPs
starting pd-server ...
```

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: examples/basic/pd-micro-service-cluster.yaml [hunk 1]
```diff
@@ -7,7 +7,10 @@ metadata:
   name: basic
   namespace: pingcap
 spec:
-  version: v7.1.1
+  startScriptVersion: v2
+  startScriptV2FeatureFlags:
+    - WaitForDnsNameIpMatch
+  version: v7.5.0
```
Comment (csuzhangxc) [inline examples/basic/pd-micro-service-cluster.yaml:13]: If we use PD ms in start script version v1, it still should work well (but may restart sometimes), right?

File: pkg/manager/member/startscript/v2/pd_start_script.go [hunk 2]
```diff
@@ -42,8 +42,8 @@ type PDStartScriptModel struct {
 
 // PDMSStartScriptModel contain fields for rendering PD Micro Service start script
 type PDMSStartScriptModel struct {
-	PDMSDomain string
-	PDMSName   string
+	PDDomain       string
```
Comment (csuzhangxc) [inline pkg/manager/member/startscript/v2/pd_start_script.go:45]: in fact, is it the domain of the real main PD process, right?

File: pkg/manager/member/startscript/v2/pd_start_script.go [hunk 3]
```diff
@@ -123,33 +123,34 @@ func renderPDMSStartScript(tc *v1alpha1.TidbCluster, name string) (string, error
...
+	m.PDDomain = fmt.Sprintf("${PD_POD_NAME}.%s.%s.svc", peerServiceName, tcNS)
 	if tc.Spec.ClusterDomain != "" {
-		m.PDMSDomain = m.PDMSDomain + "." + tc.Spec.ClusterDomain
-	}
-
-	m.PDMSName = "${PDMS_POD_NAME}"
-	if tc.AcrossK8s() || tc.Spec.ClusterDomain != "" {
-		m.PDMSName = "${PDMS_DOMAIN}"
+		m.PDDomain = m.PDDomain + "." + tc.Spec.ClusterDomain
 	}
```
Comment (csuzhangxc) [inline pkg/manager/member/startscript/v2/pd_start_script.go:131]: Do we need to handle `AcrossK8s()` anymore?

File: pkg/manager/member/startscript/v2/pd_start_script.go (hunk 2 above)
Comment (HuSharp) [reply pkg/manager/member/startscript/v2/pd_start_script.go]: just want to reuse script `pdWaitForDnsOnlySubScript`

File: pkg/manager/member/startscript/v2/pd_start_script.go (hunk 3 above)
Comment (HuSharp) [reply pkg/manager/member/startscript/v2/pd_start_script.go:131]: yes, maybe supported in future

File: examples/basic/pd-micro-service-cluster.yaml (hunk 1 above)
Comment (HuSharp) [reply examples/basic/pd-micro-service-cluster.yaml]: changed to PD&PDMS use start script v2 when enabled PDMS

File: pkg/manager/member/startscript/v2/pd_start_script.go (hunk 2 above)
Comment (HuSharp) [reply pkg/manager/member/startscript/v2/pd_start_script.go]: yes, changed

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 4]
```diff
@@ -151,6 +151,7 @@ type StartScriptVersion string
 
 const (
 	StartScriptV1 StartScriptVersion = "v1"
+	// When PD enables microservice mode, we will use start script v2.
```
Comment (csuzhangxc) [inline pkg/apis/pingcap/v1alpha1/types.go:154]: How about updating `TidbCluster.spec.StartScriptVersion`?

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**